src/fluree/db/merge/db.cljc

```diff
@@ -0,0 +1,103 @@
+(ns fluree.db.merge.db
```

Comment:
There needs to be a protocol for this functionality. A lot of the code here assumes we'll be operating on a `FlakeDB` explicitly, and there's a function that assumes that an `AsyncDB` will always wrap a `FlakeDB`. We should be able to use any dbs we receive interchangeably without ever having to explicitly check what kind of dbs they are, so new functionality on dbs should be defined in terms of new protocols.